### PR TITLE
Changes for GDD BMC-SMM server memory translation VA to PA

### PR DESCRIPTION
**Pull request by alexsmendez**
This PR handles the changes required for handling the specific memory access requirements for the BMC-SMM server.
_In includes the handling of the supported feature commands by the BMC-SMM server.
_It includes a new configuration option to mark servers that requires processing memory commands via PAs.

### CODE REVIEW

File: Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp [hunk 1]
```diff
@@ -187,8 +200,18 @@ class GdbSrvController::GdbSrvControllerImpl
...
+            strstr(pMonitorCmd, g_RequestGdbSetWritePAmode) != nullptr)
+        {
+            commandMonitor += pMonitorCmd;
+        }
+        else
+        {
+            commandMonitor += "qRcmd,";
+            commandMonitor += dataBuffer.c_str();
+            commandMonitor += dataBuffer.c_str();
+        }
```
Comment (lzybkr) [inline Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp:214]: Appending twice? If this is intentional, it deserves a comment.

File: Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp [hunk 2]
```diff
@@ -2557,6 +2609,89 @@ class GdbSrvController::GdbSrvControllerImpl
...
+        std::wstring_convert<convert_type, wchar_t> converter;
+        const std::string sGdbServerTarget = converter.to_bytes(wGdbServerTarget);
+        if (sGdbServerTarget.empty())
+        {
+            throw _com_error(E_FAIL);
+        }
+
+        pStrGdbSrvType = sGdbServerTarget.c_str();
+        gdbSrvStrTypeLength = sGdbServerTarget.length();
+        if (!cfgData.GetServerRequirePAMemoryAccess())
```
Comment (lzybkr) [inline Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp:2636]: Unused locals?

File: Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp [hunk 3]
```diff
@@ -2557,6 +2609,89 @@ class GdbSrvController::GdbSrvControllerImpl
...
+        }
+
+        messageLength = min(messageLength, C_MAX_MONITOR_CMD_BUFFER);
+        bool replyDone = false;
+        do
+        {
+            if (IsReplyOK(reply))
+            {
+                monitorResult.SetLength(monitorResult.GetLength() + messageLength);
+                memcpy(&monitorResult[monitorResult.GetLength() - messageLength], reply.c_str(), messageLength);
```
Comment (lzybkr) [inline Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp:2659]: Can SetLength fail without throwing an exception?

If so, the memcpy below isn't safe.

File: Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp (hunk 1 above)
Comment (alexsmendez) [reply Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp:213]: Good catch. The first issue will regress telemetry on exdiGDBSrv

File: Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp (hunk 2 above)
Comment (alexsmendez) [reply Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp:2631]: Yes, I don't see any VS build spew about the unused variables.

**Review by lzybkr — APPROVED** [review]

**Pull request merged**